### ISSUE TITLE
Disable animation

### ISSUE THREAD

**Issue by ShravanSunder** [proNestorAps/react-zoom-pan-pinch#32]
I am wondering if there is a way to disable animations entirely with this package? Specifically i want to disable the zoom animations

**Comment by janaagaard75** [comment]
I'm still pretty new this repo, so I don't know if this works, but does it work to set animationTime to 0? It seems that this property is available on all the different kinds of animations.

<https://pronestoraps.github.io/react-zoom-pan-pinch/?path=/story/docs-props--page>

**Comment by ShravanSunder** [comment]
It doesn't seem to turn it off at 0 for panning and recentering or zooming

**Comment by AykutSarac** [comment]
Configs doesn't seem to work for `TransformWrapper`.  The following is not working as well.

```ts
doubleClick={{
 disabled: true,
}}
```

**Comment by janaagaard75** [comment]
[The functions for manipulating the SVG](https://pronestoraps.github.io/react-zoom-pan-pinch/?path=/story/docs-handlers--page) have an `animationTime` property. Setting it to `0` disables the animation.

Does this help?

**Comment by janaagaard75** [comment]
Can you share some code, @ShravanSunder?

**Comment by ShravanSunder** [comment]
@janaagaard75 This is the component which uses this lib.  I've tried a bunch of things to turn off animation and it doesn't work.  


```
interface IFilePreviewImageProps {
  attachmentFile: IAttachmentFile;
  imageZoomControlsRef: MutableRefObject<ReactZoomPanPinchRef | null>;
  imageRotationTransform: number;
  containerDimensions: TSize;
}

/**
 * Preview of image files (mimetype begins with "image/")
 * @param props
 * @returns
 */
export const FilePreviewImage: FC<IFilePreviewImageProps> = (props) => {
  const [hasError, setHasError] = useState(false);
  const imageRef = useRef<HTMLImageElement>(null);
  const hasAspectChanged = props.imageRotationTransform % 180 != 0;
  const [imageDimensions, setImageDimensions] = useState<{
    width: string;
    height: string;
    aspectPadding: string;
  }>();

  const isMounted = useIsMounted();

  const handleImageError = (): void => {
    setHasError(true);
  };

  const handleImageLoad = (): void => {
    if (props.imageZoomControlsRef.current != null && isMounted()) {
      props.imageZoomControlsRef.current.zoomToElement(filePreviewImageId);
    }
  };

  useEffect(() => {
    // reset zoom on rotation or on window size change so that the image is centered
    if (props.imageZoomControlsRef?.current && isMounted()) {
      props.imageZoomControlsRef.current.zoomToElement(filePreviewImageId);
    }
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [
    props.imageRotationTransform,
    props.containerDimensions.height,
    props.containerDimensions.width,
  ]);

  useEffect(() => {
    const imageBoundsWidth =
      props.containerDimensions.width == null ||
      props.containerDimensions.width == 0
        ? "100%"
        : `${props.containerDimensions.width}px`;
    const imageBoundsHeight =
      props.containerDimensions.height == null ||
      props.containerDimensions.height == 0
        ? "100%"
        : `${
            props.containerDimensions.height -
            filePreviewHeaderHeight -
            filePreviewFooterHeight
          }px`;
    const imageAspectPadding = hasAspectChanged
      ? `${props.containerDimensions.width / 8 ?? 0}px`
      : "0px";
    setImageDimensions({
      width: imageBoundsWidth,
      height: imageBoundsHeight,
      aspectPadding: imageAspectPadding,
    });
  }, [
    hasAspectChanged,
    props.containerDimensions.height,
    props.containerDimensions.width,
  ]);

  useEffect(() => {
    // add a listender to image load event so that we center the image
    if (imageRef.current != null) {
      imageRef.current.removeEventListener("load", handleImageLoad);
      imageRef.current.addEventListener("load", handleImageLoad);
    }
  }, []);

  return !hasError ? (
    <TransformWrapper
      alignmentAnimation={{
        animationTime: 200,
      }}
      centerOnInit={true}
      centerZoomedOut={true}
      doubleClick={{ mode: "reset", disabled: true }}
      limitToBounds={false}
      minScale={0.1}
      panning={{ velocityDisabled: true }}
      ref={props.imageZoomControlsRef}
      velocityAnimation={{
        animationTime: 200,
      }}
      wheel={{ disabled: true }}
      zoomAnimation={{
        animationTime: 200,
      }}
    >
      {(_controls: ReactZoomPanPinchRef): ReactElement => {
        return (
          <div
            className="bills-fill-parent"
            style={{
              display: "grid",
            }}
          >
            <div
              className="bills-fill-parent"
              style={{
                placeSelf: "center",
              }}
            >
              <TransformComponent
                wrapperStyle={{
                  width: imageDimensions?.width,
                  height: imageDimensions?.height,
                }}
              >
                <div className="bills-fill-parent">
                  <img
                    alt={props.attachmentFile.name}
                    id={filePreviewImageId}
                    ref={imageRef}
                    src={XeroWebUrls.docs.getFile(props.attachmentFile.fileId)}
                    style={{
                      objectFit: "cover",
                      padding: imageDimensions?.aspectPadding,
                      transform: `rotate(${props.imageRotationTransform}deg)`,
                    }}
                    onError={handleImageError}
                  />
                </div>
              </TransformComponent>
            </div>
          </div>
        );
      }}
    </TransformWrapper>
  ) : (
    <FilePreviewLoadError />
  );
};

export default FilePreviewImage;

```

**Comment by janaagaard75** [comment]
Thanks a lot for the code. I played a bit around with the [zoom-to-element example](https://github.com/proNestorAps/react-zoom-pan-pinch/blob/main/src/stories/examples/zoom-to-element/example.tsx), and here I was able to disable the animation by adding `0` as the third parameter to the calls to `zoomToElement`, for example `zoomToElement("element1", undefined, 0);`. This is not the prettiest solution, but the best I could come up with for now.

The `zoomAnimation` and `aligmentAnimation` properties ´on the `TransformWrapper` wrapper element seem to be ignored. This is probably a bug.
